### PR TITLE
feat: route custom query support

### PR DESCRIPTION
**Pull request by bartoszherba**
## Changes

This change adds support for custom queries in the `route` method of both the `api-client` and `sdk` modules. The `route` query has also been extended with a `CategoryInterface` fragment. These changes improve the flexibility and functionality of the `route` method, allowing for more customization and control over the data returned.

### api-client

- Expanded the route query to include a `CategoryInterface` fragment.
- Introduced support for `customQueries` within the route API method.

### sdk

- Now includes support for `customQueries` in the route SDK method.
### 🔗 Linked issue

closing: [https://vsf.atlassian.net/browse/IN-3692](https://vsf.atlassian.net/browse/IN-3692)

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/api-client/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@vue-storefront/magento-api",
-  "version": "2.3.0",
+  "version": "2.2.0",
```
Comment (WojtekTheWebDev) [inline packages/api-client/package.json:3]: Are you sure that this change is propper?

**Review by WojtekTheWebDev — APPROVED** [review]
LGTM but a small question before merge

File: packages/api-client/package.json (hunk 1 above)
Comment (bartoszherba) [reply packages/api-client/package.json:3]: why not?

Comment (bartoszherba) [reply packages/api-client/package.json]: fixed

**Pull request merged**